packages/api/src/modules/push-notifications/push-notifications.service.ts

```diff
@@ -0,0 +1,77 @@
+import { Injectable, Logger } from '@nestjs/common';
+import webPush, { PushSubscription } from 'web-push';
+
+import { Role } from '../../enums/role.enum';
+import { Config } from '../config/config';
+import { PushNotificationRepository } from '../repository/push-notification.repository';
+import { SendPushNotificationDto } from './dto/send-push-notification.dto';
+import { SubscribePushNotificationDto } from './dto/subscribe-push-notification.dto';
+
+@Injectable()
+export class PushNotificationsService {
+  private logger: Logger = new Logger('PushNotifications');
```

Comment:
Usually loggers are injected as depencency injection and are singletons

Reply:
@Vatras I checked in Nest.js documentation how they recommend implementing it and basically, in the case where we don't extend Logger with our custom logic, this is a recommended way.

For clarity, I only change variable access for `readonly` and logger title to using service class name instead of a redundant string.